test/query.test.js

```diff
@@ -2702,6 +2703,34 @@ describe('query', () => {
     }
   })
 
+  test('deserialize version lambdas correctly', async () => {
+    try {
+      const rawQuery = {
+        '@query': { api_version: '3', lambda: 'X', expr: { var: 'X' } },
```

Comment:
you can remove `api_version` attribute here and send it using `client.query()` and make sure it returns `api_version: "2.12"`